rpc/rpc.go

```diff
@@ -105,17 +105,29 @@ func LinkWithSessionFilter(sessionID *string) LinkOption {
 }
 
 // NewLink will build a new request response link
-func NewLink(conn *amqp.Client, address string, opts ...LinkOption) (*Link, error) {
-	authSession, err := conn.NewSession()
+// Deprecated: use NewLinkWithContext instead.
+func NewLink(conn *amqp.Conn, address string, opts ...LinkOption) (*Link, error) {
```

Comment:
This is unfortunate.  This is a breaking change, but since `go-amqp` is still pre-release there's no way to preserve the back-compat.

Reply:
Yeah, I think we'll have to tag this as /v4, correct?

Reply:
So that's the thing.  Even if we do, there's no way to enforce that v3 remain on a compatible version of `go-amqp`.  Not saying I'm against it, but it provides little insulation (maybe the messaging is worth it though).  If we do want a /v4 then I'm going to make the other breaking changes instead of deprecating some APIs.

Reply:
Bumped to v4